src/Microsoft.OpenApi/Validations/Visitors/PathItemVisitor.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. 
+
+using System.Diagnostics;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.OpenApi.Validations.Visitors
+{
+    /// <summary>
+    /// Visit <see cref="OpenApiPathItem"/>.
+    /// </summary>
+    internal class PathItemVisitor : VisitorBase<OpenApiPathItem>
+    {
+        /// <summary>
+        /// Visit the children of the <see cref="OpenApiPathItem"/>.
+        /// </summary>
+        /// <param name="context">The validation context.</param>
+        /// <param name="pathItem">The <see cref="OpenApiPathItem"/>.</param>
+        protected override void Next(ValidationContext context, OpenApiPathItem pathItem)
+        {
+            Debug.Assert(context != null);
+            Debug.Assert(pathItem != null);
```

Comment:
We haven't been using this anywhere else. Can we be consistent and use the normal null checks?

Reply:
@Perth, 

Yes. However, it's a protected method. It's only be called from "Visit(...)" method.  So, I think it's only to add such Debug assert. 

And, yes. Maybe I should add the null check at the "Visit(...)" method.


---
In reply to: [157651170](https://github.com/Microsoft/OpenAPI.NET/pull/152#discussion_r157651170) [](ancestors = 157651170)

Reply:
protected class is practically a public class (since our customer can technically just create a class that inherits the PathItemVisitor and calls the method).

I think we should do the null check here.

---
In reply to: [157838274](https://github.com/Microsoft/OpenAPI.NET/pull/152#discussion_r157838274) [](ancestors = 157838274,157651170)